SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Globalization;
+using Content.Server.Database;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using SS14.Admin.Helpers;
+
+namespace SS14.Admin.Pages.Bans
+{
+    [Authorize(Roles = "MASSBAN")]
+    [ValidateAntiForgeryToken]
+    public class CreateMassBanModel : PageModel
+    {
+        private readonly PostgresServerDbContext _dbContext;
+        private readonly BanHelper _banHelper;
+        private readonly IWebHostEnvironment _environment;
+
+        public CreateMassBanModel(PostgresServerDbContext dbContext, BanHelper banHelper, IWebHostEnvironment environment)
+        {
+            _dbContext = dbContext;
+            _banHelper = banHelper;
+            _environment = environment;
+        }
+
+        public int BanCount { get; private set; }
+
+        public class TsvEntry
+        {
+            public string UserId { get; set; }
+            public string Address { get; set; }
+            public string Hwid { get; set; }
+            public string Reason { get; set; }
+        }
+
+        public async Task<IActionResult> OnPostAsync(IFormFile file)
+        {
+            if (file == null || file.Length <= 0)
+            {
+                ModelState.AddModelError(string.Empty, "Please select a file.");
+                return Page();
+            }
+
+            if (!file.FileName.EndsWith(".tsv", StringComparison.OrdinalIgnoreCase))
+            {
+                ModelState.AddModelError(string.Empty, "Only TSV files are allowed.");
+                return Page();
+            }
```

Comment:
You might want to check what can cause an exception (missing header, wrong columns) and if some of that doesn't throw an exception but causes issues, for example that the first row gets ignored if there is no header, you should add a check for that here

Reply:
its now checks